### PR TITLE
Run3-sim50 Define transition region for QGSP_FTFP_BERT using parameter settings

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Set the transition region among the 3 models in QGSP_FTFP_BERT using parameters defined in the cfg file

#### PR validation:

Use standard runTheMatrix script

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4Core/Application/python/g4SimHits_cfi.py [hunk 1]
```diff
@@ -150,6 +150,9 @@
         FlagBERT    = cms.untracked.bool(False),
         EminFTFP    = cms.double(3.), # in GeV
         EmaxBERT    = cms.double(6.), # in GeV
+        EminQGSP    = cms.double(12.), # in GeV
```
Comment (civanch) [inline SimG4Core/Application/python/g4SimHits_cfi.py]: @bsunanda , I do not in favor for addition of too many these parameter. From one hand, not a big deal_ from other ' these parameters are used a bit differently for 10.4 and 10.6

File: SimG4Core/PhysicsLists/plugins/QGSPCMS_FTFP_BERT.cc [hunk 2]
```diff
@@ -18,9 +19,18 @@ QGSPCMS_FTFP_BERT::QGSPCMS_FTFP_BERT(const edm::ParameterSet& p) : PhysicsList(p
   bool hadPhys = p.getUntrackedParameter<bool>("HadPhysics", true);
   bool tracking = p.getParameter<bool>("TrackingCut");
   double timeLimit = p.getParameter<double>("MaxTrackTime") * CLHEP::ns;
+  double minFTFP = p.getParameter<double>("EminFTFP") * CLHEP::GeV;
+  double maxBERT = p.getParameter<double>("EmaxBERT") * CLHEP::GeV;
+  double minQGSP = p.getParameter<double>("EminQGSP") * CLHEP::GeV;
```
Comment (civanch) [inline SimG4Core/PhysicsLists/plugins/QGSPCMS_FTFP_BERT.cc]: @bsunanda , eminQGSP is fixed in any way to 12 GeV, addition of it differently inside CMSSW is possible but only open extra freedom.

File: SimG4Core/PhysicsLists/plugins/QGSPCMS_FTFP_BERT.cc [hunk 3]
```diff
@@ -18,9 +19,18 @@ QGSPCMS_FTFP_BERT::QGSPCMS_FTFP_BERT(const edm::ParameterSet& p) : PhysicsList(p
   bool hadPhys = p.getUntrackedParameter<bool>("HadPhysics", true);
   bool tracking = p.getParameter<bool>("TrackingCut");
   double timeLimit = p.getParameter<double>("MaxTrackTime") * CLHEP::ns;
+  double minFTFP = p.getParameter<double>("EminFTFP") * CLHEP::GeV;
+  double maxBERT = p.getParameter<double>("EmaxBERT") * CLHEP::GeV;
```
Comment (civanch) [inline SimG4Core/PhysicsLists/plugins/QGSPCMS_FTFP_BERT.cc]: maxFTFP = 25 GeV in all PhysLists - we should not add this parameter - extra level of freedom without a possibility to control.

**Comment by civanch** [comment]
Hi Sunanda, I do not agree with some of solutions you propose in this PR. Ideally, we should wait until 10.6 will be integrated into CMSSW and then use proper interfaces coherent with 10.6. At the same time, you make a substantial edition, so better to integrate all this, I will try to clean it up in the next PR.

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@civanch yes i had only 10.6 in mind. please change once we shift to 10.6

**Comment by silviodonato** [comment]
+1

**Pull request merged**